src/python/WMCore/Storage/SiteLocalConfig.py

```diff
@@ -186,7 +197,7 @@ def start(*args, **kwargs):
 
 
 
-def nodeReader(node):
+def nodeReader(node,useTFC=True):
```

Comment:
Yes, I agree that the default value of ```useTFC``` is not currently consistent and is a bit confusing but actually it does not matter what is the default value of ```useTFC``` in the ```nodeReader(node,useTFC=True)``` since it is only used in ```src/python/WMCore/Storage/SiteLocalConfig.py``` and this is always called explicitly ```nodeResult = nodeReader(node,self.useTFC)```.
I have found that the only switch that is needed to chose which type of storage descriptions is used, old one storage.xml (a.k.a trivial storage description) or new one storage.json (a.k.a Rucio storage description) can be located in ```def loadSiteLocalConfig(useTFC=False)``` inside ```src/python/WMCore/Storage/SiteLocalConfig.py```. Since this ```loadSiteLocalConfig``` is always called when site local config is needed:
```src/python/WMCore/Storage/DeleteMgr.py```
```src/python/WMCore/Storage/FileManager.py```
```src/python/WMCore/Storage/StageInMgr.py```
```src/python/WMCore/Storage/StageOutMgr.py```

Therefore, I will remove the unnecessary default values of useTFC, such as the one in ```nodeReader``` you mentioned here and only keep the default value of useTFC in ```loadSiteLocalConfig(useTFC=False)``` and one can flip this to have the desired effect.

About the test of site==storageSite and volume==volume, the first one is needed to pick up the right storage.json for a storage site (if site==storageSite, the storage.json can be found "locally" where jobs are executed) and the second one is to make sure the volume in stage-out matches with one in storage.json.